lib/Runtime/Library/JavascriptPromise.h

```diff
@@ -538,28 +542,52 @@ namespace Js
         Var Resolve(Var resolution, ScriptContext* scriptContext);
         Var Reject(Var resolution, ScriptContext* scriptContext);
 
-        enum PromiseStatus
+        enum PromiseStatus : unsigned char
         {
-            PromiseStatusCode_Undefined,
-            PromiseStatusCode_Unresolved,
-            PromiseStatusCode_HasResolution,
-            PromiseStatusCode_HasRejection
+            PromiseStatusCode_Undefined     = 0x00,
+            PromiseStatusCode_Unresolved    = 0x01,
+            PromiseStatusCode_HasResolution = 0x02,
+            PromiseStatusCode_HasRejection  = 0x03
         };
 
-        bool GetIsHandled() { return isHandled; }
-        void SetIsHandled() { isHandled = true; }
-        PromiseStatus GetStatus() const { return status; }
+        static const unsigned char statusMask = 0x03;
+        static const unsigned char isHandledMask = 0x04;
```

Comment:
Instead of using masking here, is it worth making the `PromiseStatus` explicitly a 2 bit value, and then just having `isHandled` as a 1 bit value, relying on the compiler to pack the adjacent values together?

Reply:
you mean using a struct w/ explicit bit sizes?  I could do that, although I'm really tempted to just have separate fields since this is getting aligned anyway & we don't save anything for the added complexity.

Reply:
Since we weren't saving any space here due to compiler alignment, I reverted this back to separate fields since that's more clear.